### PR TITLE
[DX-763] Adding header_matches example for advanced url rewrites.

### PR DESCRIPTION
**Pull request by JRWu**
### For internal users - Please add a Jira DX PR ticket to the subject!
<br>

---

### Preview Link
When deploying docs locally, the changes can be viewed here:
[preview](https://deploy-preview-3420--tyk-docs.netlify.app/docs/nightly/transform-traffic/url-rewriting#header-match-example)

### Description
Adding advanced url rewrite example for header_matches, showcasing the notation required to reference the matched header.

#### Screenshots (if appropriate)
<br>

## Checklist

- [x] I have added a **preview link** to the PR description.
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labelled** the PR

### CODE REVIEW

**Review by firaaasss — APPROVED** [review]

File: tyk-docs/content/transform-traffic/url-rewriting.md [hunk 1]
```diff
@@ -181,6 +181,33 @@ Additionally you also mix multiple matches in the same trigger. In the example b
 ]
 ```
 
+#### Header Match Example
+You can route inbound API requests to Tyk to various based on the presence and content of a header. Suppose you have a header in the format of `store-id:1234`. Depending on the value of store-id you might wish to change the target URL.
```
Comment (dcs3spp) [inline tyk-docs/content/transform-traffic/url-rewriting.md]: _You can route inbound API requests to Tyk to various ? based on the presence and content of a header. Suppose you have a header in the format of `store-id:1234`. Depending on the value of store-id you might wish to change the target URL._

Routing to various upstream endpoints?

**Review by dcs3spp — CHANGES_REQUESTED** [review]
Hi Jia, PR looks good. Made minor suggestion.

File: tyk-docs/content/transform-traffic/url-rewriting.md (hunk 1 above)
Comment (JRWu) [reply tyk-docs/content/transform-traffic/url-rewriting.md]: Great catch, have updated the verbiage to "... to various upstream hosts or endpoints based on the presence ..."

File: tyk-docs/content/transform-traffic/url-rewriting.md [hunk 2]
```diff
@@ -181,6 +181,33 @@ Additionally you also mix multiple matches in the same trigger. In the example b
 ]
 ```
 
+#### Header Match Example
+You can route inbound API requests to Tyk to various upstream hosts or endpoints based on the presence and content of a header. Suppose you have a header in the format of `store-id:1234`. Depending on the value of store-id you might wish to change the target URL.
+
+i.e. For example, if you have the following curl request:
```
Comment (dcs3spp) [inline tyk-docs/content/transform-traffic/url-rewriting.md]:
```suggestion
For example, if you have the following curl request:
```

**Review by dcs3spp — APPROVED** [review]

**Pull request merged**